In [1]:
import lasio.examples

# Speed

In [3]:
%%timeit  -n 15

lasio.examples.open("6038187_v1.2.las")

245 ms ± 52.1 ms per loop (mean ± std. dev. of 7 runs, 15 loops each)


prior to trying to fix #217:        6038187_v1.2.las = 321 ms ± 34.8 ms per loop (mean ± std. dev. of 7 runs, 15 loops each)
read-quoted-strings branch:         6038187_v1.2.las = 378 ms ± 37.9 ms per loop (mean ± std. dev. of 7 runs, 15 loops each)
dataframe-by-generator branch:      6038187_v1.2.las = 245 ms ± 52.1 ms per loop (mean ± std. dev. of 7 runs, 15 loops each)

# Memory

In [11]:
import pickle

In [12]:
las = lasio.examples.open("logging_levels.las")
len(pickle.dumps(las))

4192151

In [13]:
las = lasio.examples.open("6038187_v1.2.las")
len(pickle.dumps(las))

198983